jerry-ext/args/arg-transform-functions.c

```diff
@@ -0,0 +1,444 @@
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  if (!jerry_value_is_function (js_arg))
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "It is not a function.");
+  }
+
+  jerry_value_t *func_p = jerryx_arg_iterator_get_dest (iterator_p);
+  *func_p = jerry_acquire_value (js_arg);
+
+  return jerry_create_undefined ();
+}
+
+jerry_value_t
+jerryx_arg_transform_function_optional (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  if (jerry_value_is_undefined (js_arg))
+  {
+    return jerry_create_undefined ();
+  }
+
+  if (!jerry_value_is_function (js_arg))
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "It is not a function.");
+  }
+
+  jerry_value_t *func_p = jerryx_arg_iterator_get_dest (iterator_p);
+  *func_p = jerry_acquire_value (js_arg);
+
+  return jerry_create_undefined ();
+}
+
+/* Object's native pointer transform function */
+
+jerry_value_t
+jerryx_arg_transform_native_pointer (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  if (!jerry_value_is_object (js_arg))
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "It is not a object.");
+  }
+
+  const jerry_object_native_info_t *expected_info_p;
+  const jerry_object_native_info_t *out_info_p;
+  expected_info_p = (const jerry_object_native_info_t *) iterator_p->c_arg_p->extra_info;
+  void **ptr_p = (void **) jerryx_arg_iterator_get_dest (iterator_p);
+  bool is_ok = jerry_get_object_native_pointer (js_arg, ptr_p, &out_info_p);
+
+  if (!is_ok || out_info_p != expected_info_p)
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "The object have no native pointer or type is not matched.");
+  }
+
+  return jerry_create_undefined ();
+}
+
+jerry_value_t
+jerryx_arg_transform_native_pointer_optional (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  if (jerry_value_is_undefined (js_arg))
+  {
+    return jerry_create_undefined ();
+  }
+
+  if (!jerry_value_is_object (js_arg))
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "It is neither a object nor undefined.");
+  }
+
+  const jerry_object_native_info_t *expected_info_p;
+  const jerry_object_native_info_t *out_info_p;
+  expected_info_p = (const jerry_object_native_info_t *) iterator_p->c_arg_p->extra_info;
+  void **ptr_p = (void **) jerryx_arg_iterator_get_dest (iterator_p);
+  bool is_ok = jerry_get_object_native_pointer (js_arg, ptr_p, &out_info_p);
+
+  if (!is_ok || out_info_p != expected_info_p)
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "The object have no native pointer or type is not matched.");
+  }
+
```

Comment:
No, no, I mean something like this:

```
static jerry_value_t jerryx_arg_transform_optional(jerryx_arg_iterator_t *iterator_p, jerryx_arg_transform_func_t transform_func)
{
	// NOTE: I added ..._peek() here, which doesn't yet exist:

	jerry_value_t peeked_js_arg = jerryx_arg_iterator_peek (iterator_p);

	if (jerry_value_is_undefined (peeked_js_arg))
	{
		jerryx_arg_iterator_pop (iterator_p);
		return jerry_create_undefined ();
	}

	return transform_func(iterator_p);
}

static jerry_value_t jerryx_arg_transform_boolean_impl(jerryx_arg_iterator_t *iterator_p)
{
	jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);

	bool to_boolean = jerry_value_to_boolean (js_arg);

	if (jerry_value_has_error_flag (to_boolean))
	{
		jerry_release_value (to_boolean);

		return jerry_create_error (JERRY_ERROR_TYPE,
					   (jerry_char_t *) "It can not be converted to a boolean.");
	}

	bool *dest = jerryx_arg_iterator_get_dest (iterator_p);
	*dest = to_boolean;

	return jerry_create_undefined ();
}

jerry_value_t
jerryx_arg_transform_boolean_optional (jerryx_arg_iterator_t *iterator_p)
{
	return jerryx_arg_transform_optional(iterator_p, jerryx_arg_transform_boolean_impl);
}

jerry_value_t
jerryx_arg_transform_boolean (jerryx_arg_iterator_t *iterator_p)
{
	return jerryx_arg_transform_boolean_impl(iterator_p);
}
```